src/SFA.DAS.LevyTransferMatching.Web/Controllers/OpportunitiesController.cs

```diff
@@ -2,24 +2,69 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using SFA.DAS.LevyTransferMatching.Web.Orchestrators;
+using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
+using System.Data;
+using SFA.DAS.Authorization.Mvc.Attributes;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Controllers
 {
-    [AllowAnonymous]
-    [Route("opportunities")]
     public class OpportunitiesController : Controller
     {
-        private readonly IOpportunitiesOrchestrator _searchFundingOrchestrator;
+        private readonly IOpportunitiesOrchestrator _opportunitiesOrchestrator;
 
         public OpportunitiesController(IOpportunitiesOrchestrator searchFundingOrchestrator)
         {
-            _searchFundingOrchestrator = searchFundingOrchestrator;
+            _opportunitiesOrchestrator = searchFundingOrchestrator;
         }
 
         public async Task<IActionResult> Index()
         {
-            var viewModel = await _searchFundingOrchestrator.GetIndexViewModel();
+            var viewModel = await _opportunitiesOrchestrator.GetIndexViewModel();
             return View(viewModel);
         }
+
+        [Route("opportunities/{encodedId}")]
+        public async Task<IActionResult> Detail(string encodedId)
+        {
+            var viewModel = await _opportunitiesOrchestrator.GetDetailViewModel(encodedId);
+
+            if (viewModel != null)
+            {
+                return View(viewModel);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost]
+        [Route("opportunities/{encodedId}")]
+        public IActionResult ConfirmOpportunitySelection(string encodedId, DetailPostRequest detailPostRequest)
+        {
+            if (!detailPostRequest.HasConfirmed.HasValue)
+            {
+                throw new DataException($"{nameof(detailPostRequest.HasConfirmed)} should be validated and have a value.");
+            }
+
+            if (detailPostRequest.HasConfirmed.Value)
+            {
+                return RedirectToAction(nameof(RedirectToApply), new { encodedId });
+            }
+            else
+            {
+                // Go back.
+                return RedirectToAction(nameof(Index));
+            }
+        }
+        
+        [DasAuthorize]
+        [Route("opportunities/{encodedId}/apply")]
+        public IActionResult RedirectToApply(string encodedId)
```

Comment:
As the next page will hang off this `/apply` route, we should probably call this something else. With the new account selection requirement, what about if we called it `SelectAccount` or similar? That would then redirect to apply if the number of accounts = 1.

Reply:
Renamed to `SelectAccount`.